src/components/Forms/AddRestock/AddRestock.js

```diff
@@ -1,9 +1,282 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+
+import LeftChevron from "components/LeftChevron/LeftChevron";
+import target from "api/api.target";
+
+//assets
+import img from "assets/tpc_logo.jpg";
+
+// styles
+import styles from "./AddRestock.module.css";
+import { Button, Grid, Paper, Typography } from "@material-ui/core";
+import InputArea from "components/InputArea/InputArea";
+import InputField from "components/InputField/InputField";
+
+// navigation
+import { useNavigation } from "contexts/NavigationContext";
+
+// products context
+import { useProducts } from "contexts/ProductsContext";
+import { useSuppliers } from "contexts/SuppliersContext";
+import ModalProducts from "components/Modal/ModalProducts";
+import ModalSuppliers from "components/Modal/ModalSuppliers";
+import Axios from "axios";
 
 function AddRestock() {
+  const { products } = useProducts();
+  const { suppliers } = useSuppliers();
+  const { goBack } = useNavigation();
+  const [openModalProducts, setOpenModalProducts] = useState(false);
+  const [openModalSuppliers, setOpenModalSuppliers] = useState(false);
+
+  // restock form
+  const [product, setProduct] = useState("");
+  const [supplier, setSupplier] = useState("");
+  const [unitCost, setUnitCost] = useState("");
+  const [quantity, setQuantity] = useState("");
+  const [costWithTax, setCostWithTax] = useState(0);
+  const [totalCost, setTotalCost] = useState(0);
+  const [notes, setNotes] = useState("");
+  const [validForm, setValidForm] = useState(false);
+
+  const submitRestockHandler = () => {
+    var restockData = {
+      product_id: parseInt(product._id),
+      supplier_id: parseInt(supplier.id),
+      quantity: parseInt(quantity),
+      unit_cost: parseFloat(unitCost),
+      cost_with_tax: parseFloat(costWithTax),
+      total_cost: parseFloat(totalCost),
+      notes: notes,
+    };
+
+    if (validForm) {
+      if (addRestock(restockData)) {
+        goBack();
+      }
+    }
+  };
+
+  const addRestock = async (formdata) => {
+    try {
+      await Axios.post(`${target}/restocks`, formdata).then((res) =>
+        console.log(res.data)
+      );
+    } catch (err) {
+      console.log(err);
+      return;
+    }
+  };
+
+  useEffect(() => {
+    const onValueChange = () => {
+      let calcWithTax = 0;
+      let totalC = 0;
+
+      if (unitCost > 0 && quantity > 0) {
+        let tax = parseFloat(unitCost) * parseFloat(0.15);
+        calcWithTax = parseFloat(unitCost) + tax;
+        totalC = parseInt(quantity) * calcWithTax;
+        setTotalCost(totalC);
+        setCostWithTax(calcWithTax);
+      }
+      return;
+    };
+
+    onValueChange();
+  }, [unitCost, quantity]);
+
+  useEffect(() => {
+    const onFormCheck = () => {
+      if (product._id && supplier.id && unitCost > 0 && quantity > 0)
+        setValidForm(true);
+    };
+    onFormCheck();
+  }, [product, supplier, unitCost, quantity]);
```

Comment:
Does it really need to check everytime there's an input change? Isn't it better to just check once you submit the form?

Reply:
Oh, I needed this one for the disabled Confirm button before anyone could submit.

Reply:
Ah, it's good then!

Reply:
Maybe, there's still another way. I think update this later.